src/Resources/skeleton/forgottenPassword/ForgottenPasswordController.tpl.php

```diff
@@ -0,0 +1,112 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace; ?>;
+
+use <?= $user_full_class_name ?>;
+use <?= $token_full_class_name ?>;
+use <?= $request_form_full_class_name ?>;
+use <?= $resetting_form_full_class_name ?>;
+use Symfony\Bundle\FrameworkBundle\Controller\<?= $parent_class_name; ?>;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+
+class <?= $class_name; ?> extends <?= $parent_class_name; ?><?= "\n" ?>
+{
+    /**
+     * @Route("/forgotten-password", name="app_forgotten_password")
+     */
+    public function request(Request $request, \Swift_Mailer $mailer): Response
+    {
+        $form = $this->createForm(<?= $request_form_class_name ?>::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user = $this->getDoctrine()->getRepository(<?= $user_class_name ?>::class)->findOneBy([
+                '<?= $email_field ?>' => $form->get('<?= $email_field ?>')->getData(),
+            ]);
+
+            // do not reveal whether an user account was found or not
+            if (!$user) {
+                return $this->redirectToRoute('app_check_email');
+            }
+
+            // generate a reset password token, that the user could use to change his password
+            $resetPasswordToken = new <?= $token_class_name?>($user);
+            $this->getDoctrine()->getManager()->persist($resetPasswordToken);
+
+            $message = (new \Swift_Message('Your password reset request'))
+                ->setFrom(['noreply@mydomain.com' => 'Noreply'])
+                ->setTo($user-><?= $email_getter ?>())
+                ->setBody($this->renderView('forgottenPassword/email.txt.twig', [
+                    'token' => $resetPasswordToken,
+                ]))
+            ;
+            $mailer->send($message);
+
+            $this->getDoctrine()->getManager()->flush();
```

Comment:
Any reasons to flush it after the email was sent instead of doing it right after we created the token?

Reply:
Not really, I will change it